subworkflows/local/deduplicate.nf

```diff
@@ -0,0 +1,167 @@
+//
+//  Carry out per-chromosome deduplication
+//
+
+include { BUILD_INTERVALS        } from '../../modules/local/build_intervals'
+include { BAM_SPLIT_BY_REGION    } from '../../subworkflows/nf-core/bam_split_by_region/main'
+include { PICARD_MARKDUPLICATES  } from '../../modules/nf-core/picard/markduplicates/main'
+include { DEDUP                  } from '../../modules/nf-core/dedup/main'
+include { SAMTOOLS_MERGE         } from '../../modules/nf-core/samtools/merge/main'
+include { SAMTOOLS_SORT          } from '../../modules/nf-core/samtools/sort/main'
+include { SAMTOOLS_INDEX         } from '../../modules/nf-core/samtools/index/main'
+include { SAMTOOLS_FLAGSTAT      } from '../../modules/nf-core/samtools/flagstat/main'
+
+workflow DEDUPLICATE {
+    take:
+    ch_bam_bai  // [ [ meta ], bam , bai ]
+    fasta       // [ [ meta ], fasta ]
+    fasta_fai   // [ [ meta ], fasta_fai ]
+
+    main:
+    ch_versions = Channel.empty()
+    ch_multiqc_files  = Channel.empty()
+    ch_refs = fasta.join(fasta_fai)
+        .map{
+            // Create additional map containing only meta.reference for combining samples and intervals
+            meta, fasta, fai ->
+            meta2 = [:]
+            meta2.reference = meta.id
+            [ meta2, meta, fasta, fai ]
+        }
+
+    // Create genomic regions file for splitting the bam before deduplication
+    BUILD_INTERVALS(fasta_fai)
+    ch_versions   = ch_versions.mix( BUILD_INTERVALS.out.versions )
+
+    // Ensure input bam matches the regions file
+    ch_bam_for_split = ch_bam_bai
+        .map{
+            // Create additional map containing only meta.reference for combining with intervals for reference
+            meta, bam, bai ->
+            meta2 = [:]
+            meta2.reference = meta.reference
+            [ meta2, meta, bam, bai ]
+        }
+        // .dump(tag:"bams_extra_meta", pretty: true)
+        .combine(
```

Comment:
Join?

Reply:
combine is needed since there are multiple samples with the same reference.